examples/debug_lines_ortho/main.rs

```diff
@@ -0,0 +1,230 @@
+        rendy::{
+            factory::Factory,
+            graph::{
+                present::PresentNode,
+                render::{RenderGroupDesc, SubpassBuilder},
+                GraphBuilder,
+            },
+            hal::{
+                command::{ClearDepthStencil, ClearValue},
+                format::Format,
+                image,
+            },
+        },
+        types::DefaultBackend,
+        Backend, GraphCreator, RenderingSystem,
+    },
+    utils::application_root_dir,
+    window::{ScreenDimensions, Window, WindowBundle},
+    winit::VirtualKeyCode,
+};
+
+struct ExampleLinesSystem;
+impl<'s> System<'s> for ExampleLinesSystem {
+    type SystemData = (
+        ReadExpect<'s, ScreenDimensions>,
+        Write<'s, DebugLines>,
+        Read<'s, Time>,
+    );
+
+    fn run(&mut self, (screen_dimensions, mut debug_lines_resource, time): Self::SystemData) {
+        let t = (time.absolute_time_seconds() as f32).cos() / 2.0 + 0.5;
+
+        let screen_w = screen_dimensions.width();
+        let screen_h = screen_dimensions.height();
+        let y = t * screen_h;
+
+        debug_lines_resource.draw_line(
+            [0.0, y, 1.0].into(),
+            [screen_w, y + 2.0, 1.0].into(),
+            Srgba::new(0.3, 0.3, 1.0, 1.0),
+        );
+    }
+}
+
+struct ExampleState;
+
+impl SimpleState for ExampleState {
+    fn on_start(&mut self, data: StateData<'_, GameData<'_, '_>>) {
+        // Setup debug lines as a resource
+        data.world.add_resource(DebugLines::new());
+        // Configure width of lines. Optional step
+        data.world
+            .add_resource(DebugLinesParams { line_width: 2.0 });
+
+        // Setup debug lines as a component and add lines to render axis&grid
+        let mut debug_lines_component = DebugLinesComponent::with_capacity(100);
+
+        let (screen_w, screen_h) = {
+            let screen_dimensions = data.world.read_resource::<ScreenDimensions>();
+            (screen_dimensions.width(), screen_dimensions.height())
+        };
+
+        (0..(screen_h as u16))
+            .step_by(50)
+            .map(f32::from)
+            .for_each(|y| {
+                debug_lines_component.add_line(
+                    [0.0, y, 1.0].into(),
+                    [screen_w, (y + 2.0), 1.0].into(),
+                    Srgba::new(0.3, 0.3, 0.3, 1.0),
+                );
+            });
+
+        (0..(screen_w as u16))
+            .step_by(50)
+            .map(f32::from)
+            .for_each(|x| {
+                debug_lines_component.add_line(
+                    [x, 0.0, 1.0].into(),
+                    [x, screen_h, 1.0].into(),
+                    Srgba::new(0.3, 0.3, 0.3, 1.0),
+                );
+            });
+
+        debug_lines_component.add_line(
+            [20.0, 20.0, 1.0].into(),
+            [780.0, 580.0, 1.0].into(),
+            Srgba::new(1.0, 0.0, 0.2, 1.0), // Red
+        );
+
+        data.world.register::<DebugLinesComponent>();
```

Comment:
I'm not sure if this `register` is required. Having debug system set up should already take care of that.

Reply:
yeap, it's good without